tests/cupy_tests/manipulation_tests/test_join.py

```diff
@@ -238,12 +239,13 @@ def test_concatenate_dtype_invalid_out(self):
         'unsafe',
     ])
     @testing.for_all_dtypes_combination(names=['dtype1', 'dtype2'])
-    @testing.numpy_cupy_array_equal(accept_error=TypeError)
+    @testing.numpy_cupy_array_equal(
+        accept_error=(TypeError, numpy.ComplexWarning))
     def test_concatenate_casting(self, xp, dtype1, dtype2, casting):
         a = testing.shaped_arange((3, 4), xp, dtype1)
         b = testing.shaped_arange((3, 4), xp, dtype1)
-        # may raise TypeError
-        return xp.concatenate((a, b), dtype=dtype2)
+        # may raise TypeError or ComplexWarning
+        return xp.concatenate((a, b), dtype=dtype2, casting=casting)
```

Comment:
OK, seems moving the warning to the re-used function was the right thing.  Just confirmed that the test fails with the fix of including `casting=casting` without the other changes.